syntax = "proto3";

message battleship {
  enum alignments {
    Ally = 0;
    Enemy = 1;
  }
  alignments alignment = 1;
  string name = 2;
  float length = 3;
  enum ship_classes {
    Corvette = 0;
    Frigate = 1;
    Cruiser = 2;
    Destroyer = 3;
    Carrier = 4;
    Dreadnought = 5;
  }
  ship_classes ship_class = 4;
  int32 crew_size = 5;
  bool armed = 6;
  message officer {
    string first_name = 1;
    string last_name = 2;
    string rank = 3;
  }
  repeated officer officers = 7;
}

message coordinates_request { float coordinates = 1; }

service ExploreSpace {
  rpc get_battleships(coordinates_request) returns (stream battleship) {};
}